utils/testutils/testutils.go

```diff
@@ -27,3 +31,37 @@ func ReadTestFile(name string) ([]byte, error) {
 		return data.Bytes(), nil
 	}
 }
+
+// GetInterface returns the best match of an interface that might be listening on a given port.
+// This is helpful when a test is being run in a CI environment under docker.
+func GetInterface(port int) string {
+	dial := func(iface string, port int) bool {
+		c, err := net.Dial("tcp", iface+":"+strconv.Itoa(port))
+		if err != nil {
+			return false
+		}
+		c.Close()
```

Comment:
One question, what about using `defer c.Close()`?

Reply:
Because there's nothing after this. So `defer` doesn't really make sense IMO.